### PR TITLE
Fix running averages for restarted simulations

### PR DESCRIPTION
**Pull request by TobiKattmann**
Hi,

When restarted at e.g. Time-Iter 2 the first value would be divided by 3 (number of timesteps counting from zero).
So if your first value in that restarted sim is a 3, the running average would be 1.

This is fixed with this commit by setting WINDOW_START_ITER = RESTART_ITER in the Config Postprocessing,
if the option is not set. If the option is set, but incorrectly, an error is raised.

## Related Work
Original implementation #836 
Change in previous behavior introduced in #1548 

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [x] My contribution is commented and consistent with SU2 style.
- [x] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: TestCases/hybrid_regression.py [hunk 1]
```diff
@@ -436,7 +436,7 @@ def main():
     square_cylinder.cfg_dir   = "unsteady/square_cylinder"
     square_cylinder.cfg_file  = "turb_square.cfg"
     square_cylinder.test_iter = 3
-    square_cylinder.test_vals = [-1.162564, 0.066401, 1.399788, 2.220402]
+    square_cylinder.test_vals = [-1.162564, 0.066401, 1.399788, 2.220402, 1.399743, 2.218605]
```
Comment (TobiKattmann) [inline TestCases/hybrid_regression.py:439]: I'll prob have to adapt this according to what the workflow says, as the available values are already slightly different to the MPI-only values and I am too lazy to set up a compilation with OMP

**Review by pcarruscag — APPROVED** [review]

File: TestCases/hybrid_regression.py (hunk 1 above)
Comment (TobiKattmann) [reply TestCases/hybrid_regression.py:439]:
```suggestion
    square_cylinder.test_vals = [-1.162564, 0.066401, 1.399788, 2.220402, 1.399743, 2.218603]
```
Tiny change to have exactly the workflow values. The Test didn't fail as it was the difference was below the threshold.

**Comment by TobiKattmann** [comment]
Thanks for the review :) Once the tests pass with this tiny commited regression value change for the hybrid case - I'll merge this in

**Comment by pcarruscag** [comment]
Sounds good thanks for the fix.

**Review by WallyMaier — APPROVED** [review]

**Pull request merged**